ipld/merkledag/merkledag.go

```diff
@@ -426,6 +438,18 @@ func sequentialWalkDepth(ctx context.Context, getLinks GetLinks, root cid.Cid, d
 		return err
 	}
 
+	// Successfully fetched "root".
+	// provide it when needed
+	if prov := options.Provider; prov != nil {
+		err := prov.Provide(ctx, root, true)
+		if err != nil && options.ErrorHandler != nil {
+			err = options.ErrorHandler(root, err)
+		}
+		if err != nil {
+			return err
+		}
```

Comment:
We probably don't want the Walk to fail because the provide operation failed.

Also don't block on `Provide()`

```suggestion
		go prov.Provide(ctx, root, true)
```